### PR TITLE
Adding concat op

### PR DESCRIPTION
**Pull request by mtopalovicTT**
#424

### CODE REVIEW

File: lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp [hunk 1]
```diff
@@ -191,6 +191,8 @@ void populateTTNNToEmitCPatterns(mlir::MLIRContext *ctx,
   patterns.add<DefaultOpConversionPattern<ttnn::TransposeOp>>(typeConverter,
                                                               ctx);
```
Comment (svuckovicTT) [inline lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp:193]: Nit: remove this line to match style.

File: include/ttmlir/Dialect/TTIR/IR/TTIROps.td [hunk 2]
```diff
@@ -360,6 +360,24 @@ def TTIR_TransposeOp : TTIR_DPSOp<"transpose"> {
...
+    let arguments = (ins Variadic<AnyRankedTensor>:$inputs,
+                         AnyRankedTensor:$output,
+                         SI32Attr:$dim,
+                         TT_OperandConstraintArrayAttr:$operand_constraints);
+
+    let results = (outs AnyRankedTensor:$result);
+
+    let extraClassDeclaration = [{
+      MutableOperandRange getDpsInitsMutable() { return getOutputMutable(); }
+    }];
```
Comment (svuckovicTT) [inline include/ttmlir/Dialect/TTIR/IR/TTIROps.td:378]: Let's add a verifier.

**Review by svuckovicTT — CHANGES_REQUESTED** [review]
I think that we need to add a verifier, otherwise looks good!

File: include/ttmlir/Dialect/TTIR/IR/TTIROps.td (hunk 2 above)
Comment (mtopalovicTT) [reply include/ttmlir/Dialect/TTIR/IR/TTIROps.td:378]: Forgot to push it...

**Review by svuckovicTT — APPROVED** [review]
Looks good to me! :)

**Pull request merged**